gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -65,8 +65,7 @@ async fn main() -> Result<(), anyhow::Error> {
         .serve_with_shutdown(listen, async {
             // Wait for plugin to signal it's shutting down
             // Shut down everything else via TaskGroup regardless of error
-            let result = plugin.join().await;
-            assert!(result.is_ok(), "{:?}", result);
+            let _ = plugin.join().await;
```

Comment:
Not sure if `expect`ing here would be useful or not. I'm ok with it. If anything panics we should normally see it anyway.